numpy/random/tests/test_seed_sequence.py

```diff
@@ -52,3 +52,29 @@ def test_reference_data():
         assert_array_equal(state, expected)
         state64 = ss.generate_state(len(expected64), dtype=np.uint64)
         assert_array_equal(state64, expected64)
+
+
+def test_zero_padding():
+    """ Ensure that the implicit zero-padding does not cause problems.
+    """
+    # Ensure that large integers are inserted in little-endian fashion to avoid
+    # trailing 0s.
+    ss0 = SeedSequence(42)
+    ss1 = SeedSequence(42 << 32)
```

Comment:
Should this test (also) spawn a single child?  Otherwise, does it actually test the trailing 0s, since the length would differ?

Reply:
This one isn't related to #16539, just explicitly testing that we did design the conversion of large integers to `uint32` words correctly. There were two obvious ways to do it, and the other (perhaps more-obvious) way would have also caused a problem.

Reply:
The way we convert `42 << 32` into a `uint32` array for consumption by `SeedSequence` converts it to `[0, 42]`. The other way to do it ("big-endian") would have converted it to `[42, 0]`, which would mean that `42`, `42 << 32`, `42 << 64` and `42 << 96` all would generate the same states.

Reply:
I relalize, I figured that in one case (if this was the "bad" version) this would be `[42,]` and in the other `[42, 0]` and I assume that this would still give a different result due to the hashing procedure being sensitive to trailing 0s as well.
But with the 0 appending from the spawn key, it becomes `[42, ..., 0]` in both cases, where `...` is filled with 0s.
In any case, I don't mind, it is still a good test.

Reply:
The problem is that the concatenated words of entropy (`entropy` and `spawn_key`, as returned by `get_assembled_entropy()`) will still get 0-padded to `pool_size` if it's shorter. It just happens implicitly.

https://github.com/numpy/numpy/blob/7a82f14075eaf8f59ea3bb2594b71639e938611e/numpy/random/bit_generator.pyx#L359-L361

The hashing algorithm is only sensitive to different number of trailing 0s for trailing 0s beyond the `pool_size`. This is testing for a potential design flaw (which I avoided) even without the `spawn_key`.